airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -297,15 +296,34 @@ def post_set_task_instances_state(dag_id, session):
         error_message = f"Task ID {task_id} not found"
         raise NotFound(error_message)
 
-    execution_date = data['execution_date']
-    try:
-        session.query(TI).filter_by(execution_date=execution_date, task_id=task_id, dag_id=dag_id).one()
-    except NoResultFound:
-        raise NotFound(f"Task instance not found for task {task_id} on execution_date {execution_date}")
+    execution_date = data.get('execution_date')
+    run_id = data.get('dag_run_id')
+
```

Comment:
should we check that either run date or exec date is passed but not both? i think i have seen that kind of
i have seen elsewhere in